modules/programs/boxxy.nix

```diff
@@ -0,0 +1,110 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.boxxy;
+
+  configPath = "${config.xdg.configHome}/boxxy/boxxy.yaml";
+  settingsFormat = pkgs.formats.yaml { };
+
+  boxxyRulesOpts = types.submodule {
+    freeformType = settingsFormat.type;
+
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = ''
+          Unique identifier of the boxxy rule. This can be any single-line string.
+        '';
+      };
+
+      target = mkOption {
+        type = types.str;
+        default = "";
+        defaultText = "~/.ssh";
+        description = ''
+          What directory/file to redirect.
+        '';
+      };
+
+      rewrite = mkOption {
+        type = types.str;
+        default = "";
+        defaultText = "$XDG_CONFIG_HOME/ssh";
+        description = ''
+          Where that file/directory should be rewritten to.
+        '';
+      };
+
+      mode = mkOption {
+        type = types.enum [ "file" "directory" ];
+        default = "directory";
+        description = ''
+          Does the current path redirect a file or a directory?
+        '';
+      };
+
+      only = mkOption {
+        type = types.listOf (types.str);
+        default = [ ];
+        example = literalExpression ''
+          [
+            "bash"
+            "/usr/bin/sh"
+          ]
+        '';
+        description = ''
+          Apply redirection ONLY to specified executable name.
+        '';
+      };
+
+      context = mkOption {
+        type = types.listOf (types.str);
+        default = [ ];
+        example = [ "/home/example/Projects/my-project" ];
+        description = ''
+          Apply redirection ONLY when in a certain directory.
+        '';
+      };
+
+      env = mkOption {
+        type = types.attrs;
```

Comment:
`types.attrs` is deprecated (cf. NixOS manual), something like `attrsOf str` should do (maybe `oneOf [ str ... ]`)

Reply:
I hope its processed the same in the json function. Going to test this change before pushing.